libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -962,8 +963,10 @@ MAV_MISSION_RESULT AP_Mission::mavlink_int_to_mission_cmd(const mavlink_mission_
         break;
 
     default:
-        // unrecognised command
-        return MAV_MISSION_UNSUPPORTED;
+        if (!stored_in_location(cmd.id)) {
```

Comment:
I'd rather see the break; approach personally. Prior art in this switch is to always break, as well as being more explicit. Otherwise if you want to move off of that then I think the empty cases should be removed now, rather then trying to split the difference.

The other path I was referring to was `mavlink_to_mission_command()` but that doesn't actually need to grow anything here.

Reply:
I looked into this further, the default should definitely be returning unsupported.

Reply:
The default is to be unsupported.  But it seems to be that anything which stores location MUST be supported, thus my patch.

But I don't need this argument.  I've changed it to a break.